httpserver/swarm.cpp

```diff
@@ -176,13 +182,33 @@ void Swarm::update_state(const all_swarms_t& swarms,
     std::copy_if(
         members.begin(), members.end(), std::back_inserter(swarm_peers_),
         [this](const sn_record_t& record) { return record != our_address_; });
-}
 
-static uint64_t hex_to_u64(const std::string& pk) {
+    // Store a copy of every node in a separate data structure
+    all_other_nodes_.clear();
 
-    if (pk.size() != 66) {
-        throw std::invalid_argument("invalid pub key size");
+    for (const auto& si : swarms) {
+        for (const auto& sn : si.snodes) {
+            if (sn != our_address_)
+                all_other_nodes_.push_back(sn);
+        }
     }
+}
+
+
+boost::optional<sn_record_t> Swarm::choose_other_node() const {
+
+    // Currently just select a node from the list of active nodes,
+    // but include decommissioned nodes in the future too
+    if (all_other_nodes_.empty())
```

Comment:
I might be overthinking it, but do we really need to keep a copy of all other nodes for (at least in this PR) only selecting a node that isn't us?
Wouldn't it be much more efficient to just choose randomly from `all_cur_swarms_` which we already have, and just retry the selection if we get unlucky and pick ourselves?

Reply:
The problem is that I want a flat array, so I can randomly select using a uniform distribution. `all_cur_swarms_` is a vector of vectors. Another valid alternative I considered is to construct a flat vector every time we need a random node, but that's somewhat inefficient (though more robust as we don't have to worry about keeping the two data structures in sync).

Reply:
It would just be 2 uniform distribution selections tho right?
First to select swarm and again to select node
But, again, might just be overthinking it and the simple flat array might be fine

Reply:
It is no longer uniform when you use two selections, it will be biased towards nodes in smaller swarms.

Reply:
Ah very true, I was actually underthinking it

Happy with this then